### DISCUSSION TITLE
Tree view using PHP 8 declaring variables in constructor does it work?

### DISCUSSION THREAD

**Discussion by fsouzaweb** [DaGhostman/vscode-tree-view#118] [General]
First of all, I want to thank you for developing it and to say that I use this extension a lot and it has worked very well...

But when declaring variables in PHP 8 inside the class constructor the extension stops working as expected the method view and variable tree, but returning the declaration to the class body works normally, this would be a bug and/or update for this one extension?

**Comment by DaGhostman** [comment]
Hey, thanks for the nice words!

Currently the extension doesn't support any of the PHP >= 8.0 features (unless something works by accident 🙂).

I am looking to find some spare time and work on it again since it is long overdue on the maintenance, but I am not really sure it is still useful, since VSCode itself added the "outline" section that does the same (with some differences).

Can you highlight why do you prefer my extension instead of the built-in functionality?

**Comment by fsouzaweb** [comment]
Please show me the print of this feature called "outline section", as I
didn't find it here as I uninstalled its extension and didn't see anything
other than the folder and file structure!

You asked me about why to use your extension, I tell you it's the best of
its kind, in fact, I don't know any other that shows the methods, variables
and other attributes of the class like yours!



Em dom., 3 de jul. de 2022 às 00:23, Dimitar Dimitrov <
***@***.***> escreveu:
